actionview/lib/action_view/lookup_context.rb

```diff
@@ -54,24 +54,17 @@ module Accessors #:nodoc:
     register_detail(:handlers) { Template::Handlers.extensions }
 
     class DetailsKey #:nodoc:
-      alias :eql? :equal?
+      @store = Concurrent::Map.new
 
-      @details_keys = Concurrent::Map.new
+      # Make hash uniq.
+      alias_method :hash, :object_id
```

Comment:
I'm not really sure, how it can give collisions, while it also uses ` name, finder.rendered_format, dependencies`.